docs/llvm/overview/index.zh.md

```diff
@@ -0,0 +1,294 @@
+
+中端的具体职责包含：
+
+* 对中间语言进行分析，如数据流分析，别名分析等。此阶段是编译器进行其他优化的基础。
+* 优化中间语言，将其表述被转化为功能等同但更快（或更小）的形式。常见的优化如内联展开，死代码消除，常量传播、循环转换等。
+
+例如我们有这么一段 LLVM IR:
+```llvm
+$ cat prog.ll
+define void @test(ptr %Q, ptr %P) {
+  %DEAD = load i32, ptr %Q
+  store i32 %DEAD, ptr %P
+  store i32 0, ptr %P
+  ret void
+}
+```
+
+我们可以使用 LLVM 提供的命令行工具对其进行一段任意的优化。通过配合 Unix 系统上的管道符，我们可以非常灵活地达到各种效果：
+```sh
+$ opt -passes=dse  prog.ll | llvm-dis -f
+; ModuleID = '<stdin>'
+source_filename = "prog.ll"
+
+define void @test(ptr %Q, ptr %P) {
+  store i32 0, ptr %P, align 4
+  ret void
+}
+```
+
+这里我们使用了 LLVM 提供的 `opt` 工具单独对 `prog.ll` 使用了死代码消除的优化。由于 `opt` 会生成人类不可阅读的 LLVM 字节码，我们接着又使用了 `llvm-dis` 将其转换为 LLVM IR。
+
+后端需要将中间代码生成对应平台的机器代码，以及进行一些 CPU 架构的细节的优化。
+
+## 如何给 LLVM 做贡献
+
+### 提交新 Bug
+
+在过去的很长一段时间里，LLVM 一直使用 [Bugzilla](https://bugs.llvm.org/) 作为其 Bug tracker，但是现在已转向 [Github issues](https://github.com/llvm/llvm-project/issues)。您可以通过查看 issue 中的 [Good Frist Issue](https://github.com/llvm/llvm-project/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) 来寻找一些容易上手的 Bug。
+
+### 提交补丁
+
+> 注意，LLVM 有意向放弃 Phabricator 并转向 Github Pull Requests，详见 [
+Code Review Process Update](https://discourse.llvm.org/t/code-review-process-update/63964)。
+
+目前，LLVM 还主要使用 [Phabricator](https://reviews.llvm.org/) 来提交补丁进行代码审查。您可以通过使用 Github 等第三方账号快速注册一个 Phabricator 账号。
+
+您可以通过使用网页或 Phabricator 自带的命令行工具 arcanist 来提交您的补丁。具体文档可参考[这里](https://llvm.org/docs/Phabricator.html#phabricator-reviews)。下面是使用 arcanist 的简单示例：
+
+> 这里我们假设您使用的是 Unix-like 的操作系统并已安装并配置好了 Git
+
+首先我们需要安装 arcanist：
+```sh
+git clone https://github.com/phacility/arcanist.git
+```
+
+接着将其添加到环境变量中：
+```sh
+export PATH="$PATH:/path/to/arcanist/bin/"
+```
+
+授权登陆：
+```sh
+arc install-certificate
+```
+
+在进行编码时，我们推荐使用 `amend` 或手动 `rebase` 将我们的代码压缩成一个 `commit`，方便后续的代码审查。
+
+当我们的 patch 在本地通过测试后，便可以准备发出去了。但在此之前一定要注意用clang-format 保证我们的 patch 符合 LLVM Coding Style。
+
+建议使用 LLVM 官方提供的 [git-clang-format](https://github.com/llvm/llvm-project/blob/main/clang/tools/clang-format/git-clang-format) 工具，效果最好，避免意外修改到其他的代码。 可以直接将 git-clang-format 所在的路径添加到 PATH 中就可以了。
+
+```sh
+git clang-format HEAD~1
+```
+
+注意format后并不会自动提交，所以您需要：
+```sh
+git commit --amend -a
+```
+
+在上述工作完成后，我们便可以使用 arc 提交补丁到 Phabricator 了：
+```sh
+arc diff
+```
+arcanist 会将最近的一个提交上传到 Phabricator 上，成功后它会给我们一个链接，这便是我们的 Revision 啦。
+
+在 reviewer 看过代码后，可能会给我们一些修改的建议，我们可以继续使用 arcanist 更新我们的补丁:
+```sh
+arc diff --update DXXXXX
+```
+其中 DXXXXX 是我们之前的 Revision，因为 arcanist 无法知道上下文，所以我们必须手动指定。
+
+### 与社区沟通
+
+* 邮件列表
+
+尽管社区已逐步淘汰了邮件列表，我们仍建议您订阅对应子项目的提交记录列表，这有助于帮助您了解项目中正在发生的事情。举个例子，如果您想向 Clang 做贡献，您可以订阅 [cfe-commits](https://lists.llvm.org/mailman/listinfo/cfe-commits) 这个列表，如果您想向 LLVM 核心做贡献，可订阅 [llvm-commits](https://lists.llvm.org/mailman/listinfo/llvm-commits) 这个列表。
+
+* [Discourse](https://discourse.llvm.org/)
+* [Discord](https://discord.com/invite/xS7Z362)
+
+### 相关链接
+
+* [Getting Started with the LLVM System](https://llvm.org/docs/GettingStarted.html)
```

Comment:
Could you provide any links to text body, and re-arrange these links as references?

Reply:
Ahh, maybe I hope that such notes are more like a paper, in the text, the references have a corresponding position. But that's fine now, I don't have a particularly strong opinion on this.

Reply:
I'm not referencing any particular text in those links but just providing some external resources that readers can dive into further.

I suspect we're good now, mark it as resolved then.